### ISSUE TITLE
Player multimedia

### ISSUE THREAD

**Issue by david42** [claroline/Distribution#146]
In a WS we have several MP3 files.
In the home of this WS, we have widgets with links to these MP3. All the players start automatically in the same time, it's a problem.

**Comment by pitrackster** [comment]
With Firefox, we have the same problems on Path with audio files added via the Rich Text Editor... 
With Chrome, the audio is muted until we press play (that can be resolved with ```<audio preload="auto"``` instead of  ```<audio preload="none"````) but will not autoplay

Looking at the html code generated by TinyMCE (when inserting an audio resource) there is an odd thing:

In the flash fallback ```<object>``` tag there is an ```<embed>``` tag with a src property AND a ```<param>``` tag inside which set the audio property.

When I remove the ```<embed>``` tag I do not have the problem anymore.

##### originally generated HTML which will autoplay
```
<p>
    <audio preload="none" controls="controls">
        <source src="/patrick/ClarolineAudioRecorder/web/app_dev.php/file/resource/media/9" type="audio/mp3"/>
        <!-- In case of the browser does not support the video tag: -->
        <object width="100%" height="400">
            <param name="audio" value="/patrick/ClarolineAudioRecorder/web/app_dev.php/file/resource/media/9"/>
            <embed src="/patrick/ClarolineAudioRecorder/web/app_dev.php/file/resource/media/9"/>
        </object>
    </audio>
    <a href="/patrick/ClarolineAudioRecorder/web/app_dev.php/resource/open/innova_audio_recorder/9">recorded-audio-201604141616</a>
</p>
```

##### HTML which will NOT autoplay
```
<p>
    <audio preload="none" controls="controls">
        <source src="/patrick/ClarolineAudioRecorder/web/app_dev.php/file/resource/media/9" type="audio/mp3"/>
        <!-- In case of the browser does not support the video tag: -->
        <object width="100%" height="400">
            <param name="audio" value="/patrick/ClarolineAudioRecorder/web/app_dev.php/file/resource/media/9"/>
<!-- HERE EMBED TAG REMOVED  -->
        </object>
    </audio>
    <a href="/patrick/ClarolineAudioRecorder/web/app_dev.php/resource/open/innova_audio_recorder/9">recorded-audio-201604141616</a>
</p>
```

On firefox and chrome (linux) this code works. But I dont use the flash fallback in both cases so I dont know if this is a real solution.

Can anyone test this with IE ?
@david42 can you see your HTML code in your case and edit it to check if the above html will make it work properly ?

**Comment by david42** [comment]
@pitrackster Thanks, it works.

Nevertheless, it's complicated for some user to edit the html.
It would be good that the players  don't start automatically.

**Comment by pitrackster** [comment]
@david42 good to know.

I think the Flash fallback is not used by any navigator agent except for IE8 (according to [can I use](http://caniuse.com/#search=audio)). Does Claroline has to support IE8 ?

I don't know your use case but in mine I think the inclusion of a Claroline Resource with TinyMCE is done by a custom code. So it will surely be possible to change this code to make it match the working HTML so that users wont have to edit the generated one.

But we have to be sure that this is a real solution. Not a [quick fix](http://www.commitstrip.com/en/2016/04/13/quick-fix/)  ;-)

**Comment by ngodfraind** [comment]
IE8 is not supported and no support will be provided.

2016-04-15 13:29 GMT+02:00 Guillou Patrick <notifications@github.com>:

> @david42 <https://github.com/david42> good to know.
>
> I think the Flash fallback is not used by any navigator agent except for
> IE8 (according to can I use <http://caniuse.com/#search=audio>). Does
> Claroline has to support IE8 ?
>
> I don't know your use case but in mine I think the inclusion of a
> Claroline Resource with TinyMCE is done by a custom code. So it will surely
> be possible to change this code to make it match the working HTML so that
> users wont have to edit the generated one.
>
> But we have to be sure that this is a real solution. Not a quick fix
> <http://www.commitstrip.com/en/2016/04/13/quick-fix/> ;-)
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/claroline/Distribution/issues/146#issuecomment-210424460>
>



-- 
Godfraind Nicolas
5, rue de la Carrière
6592 Monceau-Imbrechies
GSM: 0472 94 80 99

**Comment by pitrackster** [comment]
so we could get rid of flash fallback ! \o/

**Comment by ngodfraind** [comment]
Fixed by #154.
